From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Michael Ludwig <michaelludwig@google.com>
Date: Mon, 26 Oct 2026 11:07:50 -0400
Subject: [M158] [ganesh] Don't set texImageFormat for BGRA as RGBx

Original change's description:
> [ganesh] Don't set texImageFormat for BGRA as RGBx
>
> GrGLCaps already set fExternalTexImageFormat = 0 (disabling writes)
> when using BGRA8 texture with the RGBA8 color type. However, BGRA8 as
> RGBx8 was configured to allow writes with RGBA8 data. This is allowed
> for GL but not GLES. For consistency with BGRA-as-RGBA, this treats
> BGRA-as-RGBx the same (instead of branching on the GL standard).
>
> Bug: b/577403270
> Change-Id: I70d01796f508139b5c62b735b1e2e18e32c6d612
> Reviewed-on: https://skia-review.googlesource.com/c/skia/+/1405636
> Commit-Queue: Greg Daniel <egdaniel@google.com>
> Auto-Submit: Michael Ludwig <michaelludwig@google.com>
> Reviewed-by: Greg Daniel <egdaniel@google.com>

(cherry picked from commit ab9b798555a6cc38e7adeacb86ca6a99c5be5819)

Bug: 578797629,b/577403270
Change-Id: I70d01796f508139b5c62b735b1e2e18e32c6d612
Reviewed-on: https://skia-review.googlesource.com/c/skia/+/1407557
Auto-Submit: Chrome Cherry Picker <chrome-cherry-picker@chops-service-accounts.iam.gserviceaccount.com>
Commit-Queue: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Bot-Commit: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>

diff --git a/src/gpu/ganesh/gl/GrGLCaps.cpp b/src/gpu/ganesh/gl/GrGLCaps.cpp
index 7232d9080344fa0b5ddf25114fafcaf343ac8b1f..0a185ea345f6f5085e82f0a4d1781404a85e984d 100644
--- a/src/gpu/ganesh/gl/GrGLCaps.cpp
+++ b/src/gpu/ganesh/gl/GrGLCaps.cpp
@@ -2196,7 +2196,7 @@ void GrGLCaps::initFormatTable(const GrGLContextInfo& ctxInfo, const GrGLInterfa
                     auto& ioFormat = ctInfo.fExternalIOFormats[ioIdx++];
                     ioFormat.fColorType = GrColorType::kRGB_888x;
                     ioFormat.fExternalType = GR_GL_UNSIGNED_BYTE;
-                    ioFormat.fExternalTexImageFormat = GR_GL_RGBA;
+                    ioFormat.fExternalTexImageFormat = 0;
                     ioFormat.fExternalReadFormat = GR_GL_RGBA;
                 }
             }
